### PR TITLE
Use API key info for remote access authc effective subject

### PR DESCRIPTION
**Pull request by n1v0lg**
For the effective subject of a remote access authentication, it's a more natural model to use information associated with the remote access API key, than that of the querying cluster subject. This PR implements this change, and improves authorization failures messages for cross cluster actions.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java [hunk 1]
```diff
@@ -94,27 +94,33 @@ static String remoteActionDenied(
...
 
-        return Strings.format(
-            "action [%s] towards remote cluster [%s] is unauthorized for %s"
-                + " because no remote indices privileges apply for the target cluster",
-            action,
-            clusterAlias,
-            userText
-        );
+    private static String remoteClusterText(@Nullable String clusterAlias) {
+        return Strings.format("towards remote cluster%s ", clusterAlias == null ? "" : " [" + clusterAlias + "]");
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java:103]: I think we could improve the usefulness of the error message (and possibly in an audit context as well) if we include the remote cluster alias from the QC, and pass it along in the headers to the FC. We could store it in authc metadata, and access it here to construct an error message with the alias included. I'd do this in a separate PR; let me know what you think.

**Comment by n1v0lg** [comment]
@elasticmachine update branch bitte

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java (hunk 1 above)
Comment (ywangd) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java:103]: I think this could be useful especially when query across multiple remote clusters.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java [hunk 2]
```diff
@@ -1021,21 +1020,13 @@ public static Authentication newApiKeyAuthentication(AuthenticationResult<User>
...
-            EMPTY_ARRAY,
-            userFromRemoteCluster.fullName(),
-            userFromRemoteCluster.email(),
-            userFromRemoteCluster.metadata(),
-            userFromRemoteCluster.enabled()
-        );
         final Authentication authentication = new Authentication(
             new Subject(
-                userWithoutRoles,
+                getEffectiveSubject().getUser(),
```
Comment (ywangd) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java:1029]: Did we not have any test that asserts the effective user was the remote user? I don't see any test change that is related to the change here. If we don't have such a test, should we add one now, ie. asserting the effective user of remote access authentication is the API key user?

**Review by ywangd — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java (hunk 2 above)
Comment (n1v0lg) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java:1029]: Ya turns out this was only indirectly covered. I'll add an explicit test 👍

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Review by slobodanadamovic — APPROVED** [review]

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Comment by n1v0lg** [comment]
@elasticmachine update branch
